### PR TITLE
Add IsPropertyAvailable and IsTypeAvailable to WindowsApiChecker

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change adds IsPropertyAvailale and IsTypeAvailable methods to the WindowsApiChecker class and updates MRTK code to use them instead of directly calling ApiInformation.Is*Present().

Fixes https://github.com/microsoft/MixedRealityToolkit-Unity/issues/7426

### CODE REVIEW

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/Shared/BaseWindowsMixedRealityCameraSettings.cs [hunk 1]
```diff
@@ -102,6 +99,6 @@ private void UninitializeReprojectionUpdater()
             }
         }
 
-        #endregion IMixedRealityCameraSettings
+#endregion IMixedRealityCameraSettings
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Providers/WindowsMixedReality/Shared/BaseWindowsMixedRealityCameraSettings.cs]:
```suggestion
        #endregion IMixedRealityCameraSettings
```

File: Assets/MixedRealityToolkit/Utilities/WindowsApiChecker.cs [hunk 2]
```diff
@@ -6,7 +6,9 @@
 
 #if WINDOWS_UWP
 using Windows.Foundation.Metadata;
-#endif
+#elif (UNITY_WSA && DOTNETWINRT_PRESENT)
+using Microsoft.Windows.Foundation.Metadata;
+#endif // WINDOWS_UWP)
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/WindowsApiChecker.cs]:
```suggestion
#endif // WINDOWS_UWP
```

**Comment by sostel** [comment]
> Could you please explain in your PR description 1) what bug this is fixing 2) why this change needs to go in? For example, is it fixing a user-visible issue, or fixing some other defect? Or is this just refactoring / cleanup work?
> 
> Also, please make sure that the PR description follows [Creating a Pull Request](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/PullRequests.html#creating-a-pull-request) guidelines, specifically _"Make sure to reference any Issue / Feature Request or Task the PR relates to."_

+1 
I'm also confused what issue this is addressing. Thanks!

**Comment by davidkline-ms** [comment]
This is addressing moving code from using API contract version checks to something that is more understandable for developers. For example, what contract version added support for IsSourceKindSupported? Developers should ideally not have to remember / lookup docs to dertermine the contract when the platform will let them inquire by method / property name.

This is a cleanup task that I have discussed with @keveleigh.

**Comment by keveleigh** [comment]
@sostel https://github.com/microsoft/MixedRealityToolkit-Unity/issues/7426#issuecomment-592068483

**Pull request merged**